### PR TITLE
Refactor require calls

### PR DESCRIPTION
**Pull request by targos**
Ref: #302

A lot of tests don't pass because of the use of the rewire module.
I'm still opening this for discussion. What do we want to do about it?
- keep all the calls as var?
- convert most of the calls to const and use let for the ones that need to be mocked?
- something else?

### CODE REVIEW

**Comment by gibfahn** [comment]
So what is the issue with using `const`? Is it that rewire needs to overwrite some modules to do mocking? If so then `const` where we can and `let` otherwise sounds good (so Option 2). 

@MylesBorins I guess the alternative would be to work out whether we can get away with not using the rewire module.

**Comment by MylesBorins** [comment]
@targos I'm open to this change as long as we can keep our test suite working. Please let me know if you have any ideas as to how we could refactor the test suite to work with modules that are required using `const`. If we are not going to use `const` I don't see any benefit in using `let` over `var` for the requires

**Comment by gdams** [comment]
@targos can you look at fixing up these merge conflicts and I'll take another review

**Comment by targos** [comment]
Updated

File: lib/lookup.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 'use strict';
-const normgit = require('normalize-git-url');
-const util = require('util');
 const path = require('path');
+const util = require('util');
+
+const normgit = require('normalize-git-url');
```
Comment (gdams) [inline lib/lookup.js]: Is there a reason that you added a line here?

File: lib/update.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
 'use strict';
+const util = require('util');
+
```
Comment (gdams) [inline lib/update.js]: same here, why the line?

File: lib/lookup.js (hunk 1 above)
Comment (targos) [inline lib/lookup.js]: I kept the style of the project, that is to separate builtin, npm and local requires.

Comment (gdams) [inline lib/lookup.js]: hmm okay yeah. Can you perhaps squash the commits? I'm not sure if this needs to be 3 different commits?

Comment (gibfahn) [inline lib/lookup.js]: I think it's fine to have multiple commits while the PR is being reviewed, I assume @targos would squash once the PR was approved and ready to land (at least that's how things are often done in core).

**Comment by MylesBorins** [comment]
Before any more work is done on this we need to figure out how to fix the testing suite. Has anyone looked into this?

**Comment by gdams** [comment]
I have updated the rest of the vars to const or let and the test suite seems to be running fine: https://github.com/targos/citgm/pull/1

**Comment by MylesBorins** [comment]
hmm... I thought that rewire didn't work on `const`. Did you change offending imports to `let`?  Seems odd to be inconsistent... but if it works 
¯\\\_(ツ)\_/¯

**Comment by targos** [comment]
> Did you change offending imports to let?

@MylesBorins yes I did and I put a comment on the offending lines. For example: https://github.com/targos/citgm/blob/43641123df7843de8f415e6b9ad0416f70f6bf39/lib/citgm.js#L7

**Comment by gdams** [comment]
v4 CI: https://ci.nodejs.org/view/Node.js-citgm/job/citgm-continuous-integration/19/
v6 CI: https://ci.nodejs.org/view/Node.js-citgm/job/citgm-continuous-integration/20/ (timeout on s390x)
v7 CI: https://ci.nodejs.org/view/Node.js-citgm/job/citgm-continuous-integration/22/ (timeout on s390x)

**Comment by gdams** [comment]
@nodejs/citgm could we get some reviews please?
